code/edge/src/androidTest/java/com/adobe/marketing/mobile/NetworkResponseHandlerFunctionalTests.java

```diff
@@ -391,25 +408,34 @@ public void testProcessResponseOnError_WhenTwoEventJsonError_dispatchesTwoEvents
 		List<Event> dispatchEvents = getDispatchedEventsWith(EventType.EDGE, EventSource.ERROR_RESPONSE_CONTENT);
 		assertEquals(2, dispatchEvents.size());
 
-		Map<String, String> flattenReceivedData1 = TestUtils.flattenMap(dispatchEvents.get(0).getEventData());
-		assertEquals(5, flattenReceivedData1.size());
-		assertEquals("0", flattenReceivedData1.get("status"));
-		assertEquals("https://ns.adobe.com/aep/errors/EXEG-0201-503", flattenReceivedData1.get("type"));
-		assertEquals(
-			"Failed due to unrecoverable system error: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.commerce.purchases",
-			flattenReceivedData1.get("title")
-		);
-		assertEquals(requestId, flattenReceivedData1.get("requestId"));
-		assertEquals(event1.getUniqueIdentifier(), flattenReceivedData1.get("requestEventId"));
+		String expectedEventData1 =
+			"{" +
+			"  \"requestEventId\": \"" +
+			event1.getUniqueIdentifier() +
+			"\"," +
+			"  \"requestId\": \"" +
+			requestId +
+			"\"," +
+			"  \"status\": 0," +
+			"  \"title\": \"Failed due to unrecoverable system error: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at path $.commerce.purchases\"," +
+			"  \"type\": \"https://ns.adobe.com/aep/errors/EXEG-0201-503\"" +
+			"}";
+		JSONAsserts.assertEquals(expectedEventData1, dispatchEvents.get(0).getEventData());
 		assertEquals(event1.getUniqueIdentifier(), dispatchEvents.get(0).getParentID());
 
-		Map<String, String> flattenReceivedData2 = TestUtils.flattenMap(dispatchEvents.get(1).getEventData());
-		assertEquals(5, flattenReceivedData2.size());
-		assertEquals("2003", flattenReceivedData2.get("status"));
-		assertEquals("personalization", flattenReceivedData2.get("type"));
-		assertEquals("Failed to process personalization event", flattenReceivedData2.get("title"));
-		assertEquals(requestId, flattenReceivedData2.get("requestId"));
-		assertEquals(event1.getUniqueIdentifier(), flattenReceivedData2.get("requestEventId"));
+		String expectedEventData2 =
+			"{" +
+			"  \"requestEventId\": \"" +
+			event1.getUniqueIdentifier() +
+			"\"," +
+			"  \"requestId\": \"" +
+			requestId +
+			"\"," +
+			"  \"status\": 2003," +
+			"  \"title\": \"Failed to process personalization event\"," +
+			"  \"type\": \"personalization\"" +
+			"}";
+		JSONAsserts.assertEquals(expectedEventData2, dispatchEvents.get(1).getEventData());
```

Comment:
assertEquals also compares agains the expected right? Does it atleast compare the size (no of keys) for both expected vs actual Also we could take scope into account?

Reply:
I believe assertEquals should have 2 way equality by default and one simple addition would be to by default compare sizes to start with if we are not doing it currently.